lighthouse-core/gather/gatherers/start-url.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const ManifestGatherer = require('./manifest');
+
+class StartUrl extends ManifestGatherer {
+  afterPass(options) {
+    let startUrl = options.url;
+
+    return super.afterPass(options)
+      .then(manifest => {
+        if (!manifest) {
+          return Promise.reject(`Unable to retrieve manifest at ${options.url}`);
+        }
+
+        startUrl = manifest.value.start_url.raw;
```

Comment:
> we shouldn't test the the original url as it's an other audit that takes care of this (offline.js). I should just fail with extra information.

I think we actually need to use it.

For all error cases (missing `start_url` or invalid URL, etc), the web app manifest spec has the value of `start_url` fall back to the URL of the document instead, so I think that's what we need to do. As long as the manifest exists and is JSON, there will always be a `start_url` on the parsed result.

ManifestParser takes care of all that ([here](https://github.com/GoogleChrome/lighthouse/blob/2faa6867d03130b8331062dea8c0a7dc6938246d/lighthouse-core/lib/manifest-parser.js#L111-L144)), so this code doesn't need to handle any of this, it just might want to surface `manifest.value.start_url.debugString` as a warning if there was one